crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -16,107 +16,151 @@ use crate::{
 use quote::quote;
 use syn::{parse_quote, FnArg, Ident, Path, Result};
 
-pub fn generate_rust_signals(
-    signals: &Vec<ParsedSignal>,
-    qobject_idents: &QObjectName,
+pub fn generate_rust_signal(
+    signal: &ParsedSignal,
+    qobject_name: &Ident,
     qualified_mappings: &BTreeMap<Ident, Path>,
+    free_connect_ident: Option<CombinedIdent>,
```

Comment:
maybe there is a better way this can be done, similar to the cpp generator route?